code/modules/projectiles/guns/energy/special.dm

```diff
@@ -229,26 +225,23 @@ modules/mob/living/carbon/human/life.dm if you die, you will be zoomed out.
 	flags = NODROP
 
 /obj/item/weapon/gun/energy/kinetic_accelerator/Fire()
-	overheat = 1
-	spawn(overheat_time)
-		overheat = 0
-		recent_reload = 0
+	if(!recently_fired)

```

Comment:
mini-rant about gun code: ideally, you'd have the spawn(fire_delay) reload() take place AFTER you fire the shot, if you want to be logically consistent about it.

Problem: something stupid is handled with a sleep; probably fire_delay itself, thus resulting in obscenely long recharge times if put the self recharging after the parent call.

Our gun code is so horrible from a back-end standpoint...